.pmd/exclude.properties

```diff
@@ -0,0 +1,61 @@
+org.apache.maven.di.impl.Types=JumbledIncrementer
+org.apache.maven.execution.MavenSession=UnusedFormalParameter
+org.apache.maven.impl.DefaultPluginXmlFactory=UnusedLocalVariable
+org.apache.maven.impl.DefaultSettingsBuilder=UnusedFormalParameter
+org.apache.maven.impl.DefaultToolchainsBuilder=UnusedFormalParameter
+org.apache.maven.impl.PathSelector=CollapsibleIfStatements
+org.apache.maven.impl.model.DefaultModelInterpolator=EmptyControlStatement
+org.apache.maven.impl.model.MavenModelMerger=UselessOverridingMethod,CollapsibleIfStatements
+org.apache.maven.impl.model.ModelBuilderSessionState=CollapsibleIfStatements
+org.apache.maven.impl.resolver.LocalSnapshotMetadataGenerator=UnusedFormalParameter
+org.apache.maven.impl.resolver.scopes.Maven3ScopeManagerConfiguration=TooManyStaticImports
+org.apache.maven.impl.resolver.scopes.Maven4ScopeManagerConfiguration=TooManyStaticImports
+org.apache.maven.internal.aether.DefaultRepositorySystemSessionFactory=EmptyControlStatement
+org.apache.maven.internal.impl.DefaultLifecycleRegistry=TooManyStaticImports
+org.apache.maven.internal.impl.DefaultTypeRegistry=CollapsibleIfStatements
+org.apache.maven.internal.transformation.impl.DefaultConsumerPomArtifactTransformer=UnusedPrivateMethod
+org.apache.maven.internal.transformation.impl.DefaultConsumerPomBuilder=UnusedFormalParameter
+org.apache.maven.lifecycle.DefaultLifecycles=UnusedFormalParameter
+org.apache.maven.lifecycle.LifecycleExecutionException=UnusedFormalParameter
+org.apache.maven.lifecycle.internal.MojoExecutor=UnusedLocalVariable
+org.apache.maven.lifecycle.internal.builder.BuilderCommon=EmptyControlStatement
+org.apache.maven.lifecycle.internal.concurrent.BuildContext=UnusedLocalVariable,CollapsibleIfStatements,EmptyControlStatement
+org.apache.maven.lifecycle.internal.concurrent.BuildPlanExecutor=TooManyStaticImports
+org.apache.maven.model.io.xpp3.MavenXpp3ReaderEx=UselessOverridingMethod
+org.apache.maven.model.io.xpp3.MavenXpp3WriterEx=UselessOverridingMethod
+org.apache.maven.model.merge.MavenModelMerger=CollapsibleIfStatements
+org.apache.maven.model.merge.ModelMerger=CollapsibleIfStatements
+org.apache.maven.model.validation.DefaultModelValidator=UnusedFormalParameter
+org.apache.maven.plugin.CycleDetectedInPluginGraphException=UnusedFormalParameter
+org.apache.maven.plugin.PluginConfigurationException=UnusedPrivateField
+org.apache.maven.plugin.PluginParameterExpressionEvaluator=EmptyControlStatement
+org.apache.maven.plugin.PluginParameterExpressionEvaluatorV4=CollapsibleIfStatements,EmptyControlStatement
+org.apache.maven.project.DefaultProjectBuilder=UnusedFormalParameter
+org.apache.maven.project.MavenProject=UnusedFormalParameter,CollapsibleIfStatements
+org.apache.maven.slf4j.DefaultLogLevelRecorder=CollapsibleIfStatements,AvoidBranchingStatementAsLastInLoop
+org.apache.maven.toolchain.building.DefaultToolchainsBuilder=UnusedFormalParameter
+org.apache.maven.internal.xml.DefaultXmlService=UnusedPrivateMethod
```

Comment:
`UnusedPrivateMethod` this is why check and spot is not enough.

- https://github.com/apache/maven/pull/2310